src/ray/common/task/task_spec.h

```diff
@@ -180,6 +180,7 @@ class TaskSpecification : public MessageWrapper<rpc::TaskSpec> {
   /// Cached scheduling class of this task.
   SchedulingClass sched_cls_id_;
 
+  static std::mutex sched_mutex_;
```

Comment:
Why is this necessary?

Reply:
Because in Java worker it's possible to sumbit tasks concurrently. So we need to protect these static fields of `TaskSpecification` which could be mutated in the constructor of `TaskSpecification`.

Reply:
Theoretically, Python should also have this issue. If we haven't run into it, it's probably because we are not releasing GIL properly.

Reply:
I see, thanks! Can you use absl mutex and add thread annotations to the protected fields?